Hide the `dialog_text` property from `FileDialog`

YuriSizov opened the pull request:
It's useless in `FileDialog`, because dialogs can have either a text label or a custom layout, and `FileDialog` already has a custom layout. If you use the property, this is what it ends up looking like:


<img width="321" alt="godot windows editor dev x86_64_2023-09-11_14-47-52" src="https://github.com/godotengine/godot/assets/11782833/ef0ffc12-9563-4924-a505-6f264ac4bd83">

So to avoid confusion I propose we just hide it, like we do in other nodes in similar cases.

~PS. I'll have to rebase after https://github.com/godotengine/godot/pull/81312 due to neighboring changes in the header file.~ Done.

KoBeWi reviewed (approved):
Coincidentally I had this on my TODO list.

YuriSizov committed:
Hide the `dialog_text` property from `FileDialog`

It's useless in `FileDialog`, because dialogs can have either
a text label or a custom layout, and `FileDialog` already
has a custom layout.

akien-mga commented:
Thanks!
